src/FileIndexer.ts

```diff
@@ -148,11 +150,55 @@ export class FileIndexer {
     if (docstring.length > 0) {
       documentation.push(ts.displayPartsToString(docstring))
     }
+
     this.document.symbols.push(
-      new lsiftyped.SymbolInformation({ symbol: symbol.value, documentation })
+      new lsiftyped.SymbolInformation({
+        symbol: symbol.value,
+        documentation,
+        relationships: this.scipRelationships(declaration, symbol),
+      })
     )
   }
 
+  private scipRelationships(
+    declaration: ts.Node,
+    declarationSymbol: LsifSymbol
+  ): Relationship[] {
+    const relationships: Relationship[] = []
+    const pushImplementation = (symbol: LsifSymbol): void => {
+      if (symbol.isEmpty()) {
+        return
+      }
+      if (symbol.value === declarationSymbol.value) {
+        return
+      }
+      relationships.push(
+        new lsiftyped.Relationship({
+          symbol: symbol.value,
+          is_implementation: true,
+        })
+      )
+    }
+    if (ts.isClassDeclaration(declaration)) {
+      this.forEachParent(declaration, parent => {
+        pushImplementation(this.lsifSymbol(parent))
+      })
+    } else if (
+      ts.isMethodDeclaration(declaration) ||
+      ts.isMethodSignature(declaration) ||
+      ts.isPropertyAssignment(declaration) ||
+      ts.isPropertyDeclaration(declaration)
+    ) {
+      this.forEachParent(declaration.parent, parent => {
+        for (const member of parent.members) {
+          if (declaration.name.getText() === member.name?.getText()) {
+            pushImplementation(this.lsifSymbol(member))
+          }
+        }
+      })
```

Comment:
Is it possible for one method or property to:
1. Implement multiple members from the same parent -- If not, we should add a `return` after the `pushImplementation`.
2. Implement members from different parents -- If not, the lambda should have a return type of `"continue" | "break"` we should break the `forEachParent` loop by returning a `"break"` value from the lambda.

Small bit but maybe hoist out `declaration.name.getText()` outside the `forEachParent`?

Reply:
1. I think it's possible for a single method to implement multiple overloaded abstract methods from type definitions. For example, it's common for type definitions to provide overloads with different string literals. The JS runtime has no concept of overloading so if you have a concrete implementation of a method named `x` then it must implement all abstract method definitions in the parent types with the matching name `x`. That's my understanding at least.
2. Can you elaborate? On the JVM you can specifically override a method from a specific supertype but I don't think the JS runtime allows this.
3. Done.

Reply:
For 2., I mean that once we've found a match in some ancestor, does it make sense to keep traversing up the chain (which is what the code will do right now IIUC)? If not, we should "break" the loop in `forEachParent` by returning a result from the lambda indicating that.

Reply:
1. Done. Given that we emit the same SCIP symbol for overloaded TS methods then it doesn't make a practical difference. But at least we have a test case for this situation now.
2. I expect "Find implementations" to display all implementations, including transitive ones. For example, the implementations of `Animal#sound().` are both `Mammal#sound().` and `Animal#sound().`

```ts
interface Animal { sound: string }
class Mammal implements Animal { sound = "mammal" }
class Dog extends Mammal { sound = "woof" }
```